### PR TITLE
WORKSHOP Fix failing CLI test data unit tests to ensure CI passes

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist
* [ ] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [ ] NO REQUIRED ***unit tests*** (explain why not required)
* [ ] NO REQUIRED ***integration tests*** (explain why not required)
* [ ] NO REQUIRED ***documentation*** (explain why not required)
* [ ] Required ***release notes*** added for new/modified functionality
* [ ] Required ***updates to other repos*** complete
* [ ] NO REQUIRED ***updates to other repos*** (explain why not required)

# Related Issues
fixes NRLMMD-GEOIPS/geoips#666

# Testing Instructions
Will add testing instructions once this PR is ready for review.

# Summary
This branch fixes the remaining bugs that must be dealt with before the annual GeoIPS
Workshop commences. These bugs are largely unrelated to each other but need to be fixed
to ensure, installation, executing, and testing GeoIPS works without any hiccups. Please
add to this section below in the order in which the bugs are addressed.

1: Fixing failing unit tests so we can get the CI to pass. Unit tests
``test_geoips_config_install.py`` and ``test_log_setup.py`` had 1 or more tests that
were failing, which caused the CI to fail as well. We need to fix these unit tests so
we can actually use the CI to address problems that are coming from new PRs. Below are
the changes made to get that working.

    modified: tests/unit_tests/commandline/test_log_setup.py
    modified: tests/unit_tests/commandline/cli_top_level_tester.py
    deleted: tests/unit_tests/commandline/test_geoips_config_install.py

# Output
Will add output once this PR is ready for review.

### CODE REVIEW

File: docs/source/releases/v1_14_0a0.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.. dropdown:: Distribution Statement
+
+ | # # # This source code is protected under the license referenced at
+ | # # # https://github.com/NRLMMD-GEOIPS.
```
Comment (mindyls) [inline docs/source/releases/v1_14_0a0.rst:4]: I hope you don't mind if I start updating these one off... I'll do a wholesale change at some point, but now they bug me :-)

File: docs/source/releases/v1_14_0a0.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Version 1.14.0a0 (2024-07-12)
+*****************************
+
+* Bug Fixes
+
+  * Fix failing test data CLI unit tests
```
Comment (mindyls) [inline docs/source/releases/v1_14_0a0.rst:11]: I think these can each still be listed individually - we'll just keep reusing the branch, but we can enumerate the changes separately.

Hmm, @biosafetylvl5 that might wreak havoc on the automated release note generator?!  Not sure, depends how it is interfaced.

File: docs/source/releases/v1_14_0a0.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+Fix failing test data CLI unit tests
+------------------------------------
+
+*From GEOIPS#666: 2024-07-14, Final Bug Fixes Prior to Workshop*
+
+``test_geoips_config_install.py`` and ``test_log_setup.py`` had 1 or more tests that
+were failing, which caused the CI to fail as well. Fix these unit tests so
+we can actually use the CI to address problems that are coming from new PRs. Below are
+the files that required changes.
```
Comment (mindyls) [inline docs/source/releases/v1_14_0a0.rst:24]: YAY!

**Review by mindyls — APPROVED** [review]
Note my few changes.  Looks good to me.

**Review by mindyls — APPROVED** [review]

File: DISTRIBUTION [hunk 4]
```diff
@@ -1,11 +1 @@
 Distribution Statement A. Approved for public release. Distribution is unlimited.
```
Comment (mindyls) [inline DISTRIBUTION:1]: I put the latest DISTRIBUTION/LICENSE in here as well, to make sure the latest is available for the workshop

**Comment by mindyls** [comment]
Please merge as soon as all conversations have been resolved, and the second approval is secured.

**Review by mindyls — APPROVED** [review]

File: docs/source/releases/v1_14_0a0.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+Product families that did not have pre-defined output lists and expected data
+passed unmodified from the readers to the output formatters were failing due to
+a missing else statement ensuring the list of final_products was set even when
+the output file list was not pre-defined.  Add else statement to set final_products
+to curr_products when output file list not defined.
+
+This resolved an error with product family xrdict_area_product_to_outlist when
+it was requested without sectoring, but it worked if sectoring was requested
+(followed a different route through the code for each).
```
Comment (mindyls) [inline docs/source/releases/v1_14_0a0.rst:33]: This is the fix for Coleman's issue.

File: geoips/plugins/modules/procflows/single_source.py [hunk 6]
```diff
@@ -918,6 +918,8 @@ def process_xarray_dict_to_output_format(
             xobjs["METADATA"],
             area_def=area_def,
         )
+    else:
+        final_products = curr_products
```
Comment (mindyls) [inline geoips/plugins/modules/procflows/single_source.py:922]: This is the fix for Coleman's issue.

File: docs/source/releases/v1_14_0a0.rst (hunk 5 above)
Comment (mindyls) [reply docs/source/releases/v1_14_0a0.rst:33]: I meant for this to be on its own branch :-(

Comment (mindyls) [reply docs/source/releases/v1_14_0a0.rst:33]: moved this to its own branch.  Likely will autoclose the other PR if this one gets merged first.  Will clear out these changes from here if the other one gets merged first.

File: geoips/plugins/modules/procflows/single_source.py (hunk 6 above)
Comment (mindyls) [reply geoips/plugins/modules/procflows/single_source.py:922]: Note other PR opened.

File: tests/unit_tests/commandline/test_log_setup.py
Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:1]: This looks good to me for now. In the long-run, though, I would like to remove anything that actually requires reading a data file from the unit tests unless we can create some miniature files that would be representative. I would call this an integration test. If there are things in the readers that need to be tested, we should split the readers up into smaller, testable functions.

**Review by jsolbrig — APPROVED** [review]

File: tests/unit_tests/commandline/test_log_setup.py
Comment (evrose54) [reply tests/unit_tests/commandline/test_log_setup.py:1]: Agreed. Off the top of my head, the only other unit test that deals with data files is ``geoips run``. Thinking about it now, it makes sense to remove that unit test, as the integration tests are technically running exactly that. They call scripts which make use of ``geoips run``, so we can likely remove this unit test altogether.

Comment (jsolbrig) [reply tests/unit_tests/commandline/test_log_setup.py:1]: Great, thank @evrose54. And thanks for creating the issues needed to address this.

Comment (mindyls) [reply tests/unit_tests/commandline/test_log_setup.py:1]: Agreed.

File: docs/source/releases/v1_14_0a0.rst (hunk 2 above)
Comment (mindyls) [reply docs/source/releases/v1_14_0a0.rst:11]: Meaning - I don't think we should just have a single "Bug fixes for workshop" header with all the changes under that single header.  We should, as normal, create separater PRs with separate sections in the release notes for each change we make.  We'll just reuse the Issue to create more branches.  It is how I would liek it now, so if you agree go ahead and resolve.

**Pull request merged**